src/transaction.ts

```diff
@@ -223,6 +223,11 @@ export class Transaction implements ISignable {
     return this.chainID;
   }
 
+  setChainID(chainID: ChainID) {
+    this.chainID = chainID;
+    this.doAfterPropertySetter();
```

Comment:
I think this is a side effect that the user does not expect - I remember debugging with @arhtudormorar quite a lot what was the problem with a transaction, and it was missing the signature after setting some properties. 
If a user changes something on a signed transaction, the signature will be invalid even if it's set or if it's empty, so I don't know if we should force the reset for them

Reply:
Fixed, removed `doAfterPropertySetter()`.